src/corona_hakab_model/generation/circles_generator.py

```diff
@@ -104,8 +109,37 @@ def __init__(
         self.social_circles_by_agent_index = {}
         self.fill_social_circles_by_agent_index()
 
+        self.num_of_random_connections = self.generate_random_connections()
+
         self._fill_population_data()
 
+    def generate_random_connections(self):
+        num_of_random_connections = np.zeros((len(self.agents), len(ConnectionTypes)), dtype=float)
+        for connection_type in connection_types.With_Random_Connections:
+            exp_mean = self.circles_consts.random_connections_dist_mean[connection_type]
+
+            for circle in self.social_circles_by_connection_type[connection_type]:
+                num_of_agents_in_circle = len(circle.agents)
+                if num_of_agents_in_circle == 1:
+                    # No random connections
+                    continue
+
+                agents_id = [a.index for a in circle.agents]
+
+                # Sample from exponential distribution
+                rand_connections = np.random.exponential(exp_mean, len(agents_id))
+
+                # You can't have more random connections than the number of people (other than you) in the circle
+                rand_connections[rand_connections > num_of_agents_in_circle - 1] = num_of_agents_in_circle - 1
```

Comment:
Depending on the size of the array, np.clip() would prob. be better (but, it does create a new array so...)

Reply:
forgot about `np.clip()`. thx.